controlplane/kubeadm/internal/workload_cluster_coredns_test.go

```diff
@@ -282,6 +289,105 @@ kind: ClusterConfiguration
 			objs:          []client.Object{depl, cm, kubeadmCM},
 			expectErr:     false,
 			expectUpdates: true,
+			expectImage:   "k8s.gcr.io/some-repo/coredns:1.7.2",
```

Comment:
Might have missed it. Should we add another test case for the upgrade from 1.7 to 1.8.0 for `k8s.gcr.io/some-repo`?

Reply:
Test added, PTAL

Reply:
Should we use `v1.8.0` as tag instead of `1.8.0` here? But lgtm as is for me

Reply:
internally we are using parse tolerant, so it does not make different for the upgrade logic/for the test (it makes a difference when you are hitting a real image repository thought, because image tag must match)

Reply:
Yup thought something like that,